homeassistant/components/touchline/__init__.py

```diff
@@ -1 +1,60 @@
-"""The touchline component."""
+"""Platform for Roth Touchline floor heating controller."""
+from __future__ import annotations
+
+import logging
+import re
+
+from pytouchline import PyTouchline
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Roth Touchline from a config entry."""
+    host = entry.data[CONF_HOST]
+    # Remove HTTPS and HTTP from URL
+    pattern = "https?://"
+    host = re.sub(pattern, "", host)
+    host = "http://" + host
```

Comment:
Home Assistant comes with [`yarl`](https://github.com/aio-libs/yarl), which makes this a lot easier:

```python
from yarl import URL

url = URL(entry.data[CONF_HOST])
host = url.with_scheme("http") 
```

Reply:
Yes, if the user inputs an URL with http:// or similar first. But I wanted to accept entries without a scheme, so input could be http://10.0.0.1 or 10.0.0.1 or maybe srv.myhost.com. I have looked into https://yarl.aio-libs.org/en/latest/api.html and found that
`url = URL('10.0.1.6').host`
returns 'None'.

But thanks for looking into it, I don't see it done any easier way?

Reply:
Got it—that makes sense. 👍🏻